text/trace/0107-sampling-score.md

```diff
@@ -0,0 +1,217 @@
+# Associating sampling score with the trace
+
+Enable consistent sampling across distributed application with different
+sampling rates and probability calculation algorithms.
+
+## TL;DR
+
+**Score** is a floating point number associated with
+the trace. It's calculated when trace starts and flows in the `tracestate`,
+it's used by samplers to make consistent sampling decisions.
+
+Service that starts the trace calculates the score and adds it to the
+`tracestate` so downstream services can re-use it to make their sampling
+decisions *instead of* re-calculating score as a function of trace-id
+(or trace-flags).
+
+*Score* is not related to sampling *rate* (aka *probability* which represents
+sampler's configuration not specific to trace).
+
+## Motivation
+
+The goal is to enable a mechanism for consistent (best effort) sampling
+between services with different sampling rates and different probability
+calculation algorithms (for interoperability with existing tracing tools).
+
+Consistent sampling decision made in each app of a distributed trace is
+important for better user experience of trace analysis. Consistency is achieved
+by following means:
+
+1. Same hashing algorithms used across all apps in a trace.
+   Coordination of sampling algorithms across multiple apps not always possible:
+   for example existing components in a system use vendor-specific
+   tracing tool (pre-OpenTelemetry and update is hard to justify) while there
+   is a desire to use OpenTelemetry for new components.
+
+2. Sampling flag propagated from the head component/app is used by downstream
+   apps to sample in a given trace.
+   It requires to trust upstream decision and does not allow to have different
+   sampling rates across different components.
+
+## Explanation
+
+Sampling propabaility is generated by the first service to make sampling
+decision. It's a random float number in [0, 1] range.
+Score is stamped on the span and also propagated further within `tracestate`.
```

Comment:
probably we need a sampler name + sampler details. So here would be "ExternalScoreSampler(0.5 /*probabilty*/, 0.12343 /*score*/)"

Reply:
I don't see why we'd need the sampler details on the tracestate. What scenario are you thinking about?